Classes/DDFileLogger.m

```diff
@@ -1531,3 +1544,101 @@ BOOL doesAppRunInBackground() {
 }
 
 #endif
+
+static NSUInteger kMaximumBytesCountInBuffer = (1 << 10) * (1 << 10); // 1 MB.
+static NSUInteger kDefaultBytesCountInBuffer = (1 << 10);
+@interface DDFileLoggerWithBuffer : DDFileLogger
+@property (assign, nonatomic, readwrite) NSUInteger maximumBytesCountInBuffer;
+@end
+
+@interface DDFileLoggerWithBuffer() {
+    NSOutputStream *_bufferStream;
+    NSUInteger _bufferSize;
+}
+@end
+
+@interface DDFileLoggerWithBuffer (StreamManipulation)
+- (void)flushBuffer;
+- (void)dumpBufferToDisk;
+- (void)appendToBufferData:(NSData *)data;
+@end
+
+@implementation DDFileLoggerWithBuffer (StreamManipulation)
+- (void)flushBuffer {
+    // do something.
+    [_bufferStream close];
+    _bufferStream = nil;
+    _bufferSize = 0;
+}
+- (void)dumpBufferToDisk {
+    // do something.
+    __auto_type data = (NSData *)[_bufferStream propertyForKey:NSStreamDataWrittenToMemoryStreamKey];
+    [super logData:data];
```

Comment:
Why not `[self logData:data];`?

Reply:
@ffried 
I want to patch `logData` to decide which channel ( buffer or disk ) are preferable at this moment.
Here I have a mistake.

`logData` should be:

```objective-c
- (void)logData:(NSData *)data {
    [self putDataIntoBufferOrToDisk:data];
}
```

And `[super logData:data]` will do what it intended to: it put data to disk.

Reply:
Still it's kind of a "misuse" of `super`.
It's not clear (at least to me) when reading this code, that the super call effectively means "write data to disk".
But I see your intention. How about something like this?
```objc
- (void)logData:(NSData *)data {
    if ([self shouldDumpBufferToDisk]) {
        __auto_type bufferedData = (NSData *)[_bufferStream propertyForKey:NSStreamDataWrittenToMemoryStreamKey];
        [super logData:bufferedData]; // Write data to disk
        [self flushBuffer];
    }
    [self appendToBufferData:data];
}
```

`- (void)dumpBufferToDisk` and `- (void)putDataIntoBufferOrToDisk:(NSData *)data` would be removed completely and the `super` call would be inside the correct method.

Also, we only have the line `[self appendToBufferData:data]` once (instead of once in `if` and once in `else`)